pkg/recordset/recordset.go

```diff
@@ -318,7 +409,11 @@ func targetStackName(clusterName string) string {
 	return fmt.Sprintf(targetStackNameFmt, clusterName)
 }
 
-func extractClusterName(sourceStackName string) string {
+func extractClusterName(sourceStackName string) (string, error) {
 	parts := strings.Split(sourceStackName, "-")
-	return parts[1]
+	if len(parts) >= 2 {
+		return parts[1], nil
+	}
+
+	return "", invalidClusterNameError
```

Comment:
I think it would be good to annotate the error with the actual cluster name.

Reply:
fixed